src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -122,9 +122,32 @@ public class DataBlockService : IDataBlockService
                 {
                     var viewModel = _mapper.Map<DataBlockViewModel>(dataBlock);
 
+                    var subjectId = dataBlock.Query.SubjectId;
+
+                    var releaseContentBlock = await _context.ReleaseContentBlocks
+                        .Include(rcb => rcb.Release)
+                        .SingleOrDefaultAsync(rcb => rcb.ContentBlockId == id);
+
+                    var releaseId = releaseContentBlock.ReleaseId;
```

Comment:
That translates into a huge select statement selecting all the columns of Release joined with all the columns of ReleaseContentBlocks, just to get the release id.

You could rewrite that as

```
var releaseId = await _context.ReleaseContentBlocks
  .Where(rcb => rcb.ContentBlockId == id)
  .Select(rcb => rcb.ReleaseId)
  .SingleAsync();
```

which should result in a simple select statement `SELECT releaseId FROM ReleaseContentBlocks WHERE ContentBlockId = <id>`.